packages/asset-kit-block/src/components/AssetGrid.tsx

```diff
@@ -37,7 +38,8 @@ export const AssetGrid = ({
                     {currentAssets
                         ? currentAssets.map((asset) => (
                               <ThumbnailItem
-                                  key={asset.id}
+                                  key={`${asset.id}-${generateRandomId()}`}
```

Comment:
Why is this needed? As I think this will generate a new ID each rerender => it will remount the Thumbnail item each rerender which might not be what you want. Could you use the blockId here or the [React useID() hook](https://react.dev/reference/react/useId) which creates IDs that persist across rerenders?

Reply:
Good catch its not needed anymore, did first that and then saw what the real problem is, that the same asset could be added multiple times